packages/office-ui-fabric-react/src/components/Calendar/CalendarYear.tsx

```diff
@@ -0,0 +1,369 @@
+import * as React from 'react';
+import { KeyCodes, css, getRTL } from '../../Utilities';
+import { ICalendarIconStrings } from './Calendar.types';
+import { FocusZone } from '../../FocusZone';
+import * as stylesImport from './Calendar.scss';
+import { Icon } from '../../Icon';
+const styles: any = stylesImport;
+
+const CELL_COUNT = 12;
+
+export interface ICalendarYearRange {
+  fromYear: number;
+  toYear: number;
+}
+
+export interface ICalendarYearRangeToString {
+  (range: ICalendarYearRange): string;
+}
+
+export interface ICalendarYearStrings {
+  prevRangeAriaLabel?: string | ICalendarYearRangeToString;
+  nextRangeAriaLabel?: string | ICalendarYearRangeToString;
+}
+
+export interface ICalendarYear {
+  focus(): void;
+}
+
+export interface ICalendarYearProps {
+  navigationIcons?: ICalendarIconStrings;
+  navigatedYear?: number;
+  selectedYear?: number;
+  minYear?: number;
+  maxYear?: number;
+  onSelectYear?: (year: number) => void;
+  onRenderTitle?: (props: ICalendarYearHeaderProps) => React.ReactNode;
+  className?: string;
+  strings?: ICalendarYearStrings;
+}
+
+const DefaultCalendarYearStrings: ICalendarYearStrings = {
+  prevRangeAriaLabel: 'Previous Year Range',
+  nextRangeAriaLabel: 'Next Year Range'
+};
+
+const DefaultNavigationIcons: ICalendarIconStrings = {
+  leftNavigation: 'Up',
+  rightNavigation: 'Down',
+  closeIcon: 'CalculatorMultiply'
+};
+
+interface ICalendarYearGridCell {
+  focus(): void;
+}
+
+interface ICalendarYearGridCellProps {
+  year: number;
+  current?: boolean;
+  selected?: boolean;
+  disabled?: boolean;
+  onSelectYear?: (year: number) => void;
+}
+
+class CalendarYearGridCell extends React.Component<ICalendarYearGridCellProps, {}> implements ICalendarYearGridCell {
+  private _buttonRef: HTMLButtonElement;
+  public focus() {
+    if (this._buttonRef) {
+      this._buttonRef.focus();
+    }
+  }
+  public render() {
+    const { year, current, selected, disabled, onSelectYear } = this.props;
+    return (
+      <button
+        className={css('ms-DatePicker-yearOption', styles.yearOption, {
+          ['ms-DatePicker-year--current ' + styles.yearIsCurrentYear]: current,
+          ['ms-DatePicker-day--highlighted ' + styles.yearIsHighlighted]: selected,
+          ['ms-DatePicker-yearOption--disabled ' + styles.yearOptionIsDisabled]: disabled
+        })}
+        type="button"
+        role="gridcell"
+        onClick={!disabled && onSelectYear ? this._onClick : undefined}
+        onKeyDown={!disabled && onSelectYear ? this._onKeyDown : undefined}
+        disabled={disabled}
+        aria-label={String(year)}
+        aria-selected={selected}
+        ref={this._onButtonRef}
+      >
+        {year}
```

Comment:
It is possible that in some languages (Japanese, for example) we might need to add characters around the number.
If that is the case, it might be interesting to have a callback to format the year value.
There is one available in ICalendarFormatDateCallbacks.formatYear

Reply:
:thumbsup: